### DISCUSSION TITLE
Any way to invalidate the cache?

### DISCUSSION THREAD

**Discussion by olalofgren** [ThatGuySam/vimeo-thumbnail-server#43] [General]
When changing the thumbnail on the original video, the thumbnail stays in the cache for a long time.
Is there any way to invalidate the cache for a single video?

**Comment by ThatGuySam** [comment]
You should be able to just add a new query string like: ?v2

Just don't abuse it or I'll have to turn on query string caching for
everyone

**Reply by Ben-Atherton** [reply]
I'm also looking for a solution to this and have tried adding ?v2 to the URL but to no avail.

An example is https://vumbnail.com/837861355.jpg?v2

**Reply by ThatGuySam** [reply]
What should the updated thumbnail look like? This is what I'm seeing on the Vimeo page for it: 

<img width="1481" alt="Screenshot 2023-09-15 at 5 33 17 PM" src="https://github.com/ThatGuySam/vumbnail/assets/2830379/7cd8902f-a489-4984-a5b7-5b256f8353e2">

**Reply by Ben-Atherton** [reply]
The updated thumbnail should have the logo in the bottom left corner which I can see it now has. How long are images usually cached for?

**Reply by ThatGuySam** [reply]
As long as possible to keep the platform scalable and images serving fast, but I think Cloudflare maxes out at 30 days. 

It also depends on things like device, ISP, and region. 

I'm considering a paid option for features like shorter cache times or manual cache resets, but I haven't seen enough demand yet.

**Reply by numeralsix** [reply]
Invalidate cache feature would be great. ?refresh=1

Thanks for maintaining this.
